internal/server/data/migrations.go

```diff
@@ -167,7 +166,25 @@ func addAuthURLAndScopeToProviders() *migrator.Migration {
 // #2360: delete duplicate grants (the same subject, resource, and privilege) to allow for unique constraint
 func deleteDuplicateGrants() *migrator.Migration {
 	return &migrator.Migration{ID: "202207081217", Migrate: func(tx *gorm.DB) error {
-		return deleteDuplicates(tx, "subject, resource, privilege", &models.Grant{})
```

Comment:
I'm not sure why, but this `deleteDuplicates` was not working. The test showed that duplicate records still existed.  I replaced it with raw SQL that I believe matched the original intent, and it works now. So I'm guessing gorm was building the wrong query.

Reply:
This is the query that I had written out but there was some issues when translating it gorm I guess,

Reply:
This is the query that I had written out but there was some issues when translating it gorm I guess,